app/vmui/packages/vmui/src/components/ExploreAlerts/helpers.ts

```diff
@@ -18,3 +19,20 @@ export const formatEventTime = (raw: string) => {
   const t = dayjs(raw);
   return t.year() <= 1 ? "Never" : t.format("DD MMM YYYY HH:mm:ss");
 }
+
+export const getStates = (rule: Rule) => {
+  const output: Record<string, number> = {};
+  const alertsCount = rule?.alerts?.length || 0;
+  if (alertsCount > 0) {
+    rule.alerts.forEach((alert) => {
+      if (alert.state in output) {
+        output[alert.state] += 1;
+      } else {
+        output[alert.state] = 1;
+      }
+    });
+  } else {
+    output[rule.state] = 1;
+  }
+  return output;
+};
```

Comment:
This can be simplified using `reduce`:

```suggestion
export const getStates = (rule: Rule): Record<string, number> => {
  if (!rule.alerts?.length) {
    return { [rule.state]: 1 };
  }

  return rule.alerts.reduce((acc, alert) => {
    acc[alert.state] = (acc[alert.state] ?? 0) + 1;
    return acc;
  }, {} as Record<string, number>);
};
```